tests/chainerx_tests/unit_tests/routines_tests/test_connection.py

```diff
@@ -28,38 +30,64 @@ def _create_conv_args(
-    ((1, 3, 4, 4), (2, 3, 3, 3), (2,), 2, (2, 0)),
-    ((2, 3, 4, 4), (2, 3, 3, 3), None, 2, (2, 0)),
-    ((1, 3, 2, 6, 3), (2, 3, 1, 3, 2), (2,), 2, (2, 0, 1)),
-    ((1, 3, 2, 6, 3), (2, 3, 1, 3, 2), (2,), (1, 2, 3), (2, 0, 1)),
-    ((2, 3, 2, 6, 3), (2, 3, 1, 3, 2), None, (1, 2, 3), (2, 0, 1)),
-])
+@chainer.testing.parameterize(*(
+    # without bias
+    chainer.testing.product([
+        chainer.testing.from_pytest_parameterize(
+            'x_shape,w_shape,b_shape,stride,pad', [
+                ((1, 3), (5, 3), None, 1, 0),
+                ((1, 3, 4), (5, 3, 2), None, 3, 2),
+                ((2, 3, 4, 4), (2, 3, 3, 3), None, 2, (2, 0)),
+                ((2, 3, 2, 6, 3), (2, 3, 1, 3, 2), None, (1, 2, 3), (2, 0, 1)),
+            ]),
+        chainer.testing.from_pytest_parameterize(
+            'in_dtypes,out_dtype', dtype_utils.result_dtypes_two_arrays)
+    ]) +
+    # with bias
+    chainer.testing.product([
+        chainer.testing.from_pytest_parameterize(
+            'x_shape,w_shape,b_shape,stride,pad', [
+                ((1, 3), (5, 3), (5,), 1, 0),
+                ((2, 3, 4), (5, 3, 1), (5,), 1, 0),
+                ((1, 3, 4), (5, 3, 2), (5,), 3, 2),
+                ((2, 3, 4, 4), (2, 3, 3, 3), (2,), 1, 0),
+                ((1, 3, 4, 4), (2, 3, 3, 3), (2,), (1, 2), 1),
+                ((1, 3, 4, 4), (2, 3, 3, 3), (2,), 2, (2, 0)),
+                ((1, 3, 2, 6, 3), (2, 3, 1, 3, 2), (2,), 2, (2, 0, 1)),
+                ((1, 3, 2, 6, 3), (2, 3, 1, 3, 2), (2,), (1, 2, 3), (2, 0, 1)),
+            ]),
+        chainer.testing.from_pytest_parameterize(
+            'in_dtypes,out_dtype', dtype_utils.result_dtypes_three_arrays)
+    ])
+))
 @chainer.testing.parameterize_pytest('cover_all', [True, False])
 class TestConv(op_utils.ChainerOpTest):
 
-    def setup(self, float_dtype):
+    def setup(self):
+        if len(self.in_dtypes) == 3:
+            x_dtype, w_dtype, b_dtype = self.in_dtypes
+        else:
+            (x_dtype, w_dtype), b_dtype = self.in_dtypes, None
+
+        x_kind = numpy.dtype(x_dtype).kind
+        w_kind = numpy.dtype(w_dtype).kind
+        b_kind = None if b_dtype is None else numpy.dtype(b_dtype).kind
 
         device = chainerx.get_default_device()
         if device.backend.name == 'cuda' and len(self.x_shape) <= 3:
             # TODO(hvy): Support 1 dimensional convolution with CUDA.
             pytest.skip('cudnn does not support 1-dim convolution')
         if device.backend.name == 'cuda' and self.cover_all:
             pytest.skip('cudnn does not support cover_all')
-        if device.backend.name == 'native' and float_dtype == 'float16':
-            # TODO(niboshi): Fix accuracy
-            pytest.skip('Native float16 operation has insufficient accuracy')
 
-        self.dtype = float_dtype
+        # Skip backward/double-backward tests for int dtypes
+        if (x_kind != 'f' and w_kind != 'f'
+                and (b_kind is None or b_kind != 'f')):
```

Comment:
The condition `b_kind is None` is not needed.

Reply:
It's intentional.
I want to make it explicit.